src/blog/templates/blog/includes/comments.html

```diff
@@ -181,6 +198,60 @@ <h2 class="text-2xl">Comments</h2>
         });
         li.appendChild(likeButton);
 
+        likeButton.addEventListener("mouseover", function () {
+            const commentId = comment.id
+            fetch(`/blog/api/comment-likes/${commentId}/`, {
+                method: "GET",
+                headers: {
+                    "Content-Type": "application/json",
+                    "X-CSRFToken": document.querySelector(
+                        'input[name="csrfmiddlewaretoken"]',
+                    ).value,
+                },
+            })
+            .then(response => response.json())
+            .then(data => {
+                const likeList = document.getElementById(`comment-like-list-${commentId}`);
+                const loggedUser = document.getElementById("user-info");
+                const loggedInUser = loggedUser.attributes.user.textContent;
+
+                function addListItem (like) {
+                    var newLi = document.createElement('li');
+                    if (like.first_name && like.last_name) {
+                        newLi.appendChild(document.createTextNode(`${like.first_name} ${like.last_name}`))
+                    } else if (like.first_name) {
+                        newLi.appendChild(document.createTextNode(like.first_name))
+                    } else if (like.last_name) {
+                        newLi.appendChild(document.createTextNode(like.last_name))
+                    } else if (like.username) {
+                        newLi.appendChild(document.createTextNode(like.username))
+                    } else {
+                        newLi.appendChild(document.createTextNode(like))
+                    }
+                    newLi.className= 'text-sm'
+                    newLi.id = `comment-${commentId}-${like.username}`
+                    likeList.appendChild(newLi);
+                }
+
+                if (likeList.children.length === 0) {
+                    for (like of data.liked_by) {
+                        addListItem(like)
+                    }
+                } else if (data.like_count > likeList.children.length) {
+                    addListItem(loggedUser.attributes.display.textContent);
+                } else if (data.like_count < likeList.children.length) {
+                    var li = document.getElementById(`comment-${commentId}-${loggedInUser}`)
+                    likeList.removeChild(li);
+                }
```

Comment:
Instead of comparing the like count with the children, and adding / removing accordingly, you can also just remove all children and then add all of them according to the data. Way less error prone.

See also the `renderComments` function, which just sets `innerHTML` to an empty string and subsequently renders every individual comment. It's way easier :D

https://github.com/drikusroor/brazil-blog/blob/d5c3cb36d0ce97c7433588d6f865e9e6e9e4f82e/src/blog/templates/blog/includes/comments.html#L304-L309